lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -354,3 +356,23 @@ def disable(self) -> OpenSearchPluginConfig:
     def name(self) -> str:
         """Returns the name of the plugin."""
         return "opensearch-knn"
+
+
+class OpenSearchPrometheusExporter(OpenSearchPlugin):
+    """Implements the opensearch-knn plugin."""
+
+    @property
+    def name(self) -> str:
+        """Returns the name of the plugin."""
+        return "prometheus-exporter"
+
+    def config(self) -> OpenSearchPluginConfig:
+        """Returns a plugin config object to be applied for enabling the current plugin."""
+        return OpenSearchPluginConfig(
+            config_entries_to_add={
+                "prometheus.metric_name.prefix": "opensearch_",
+                "prometheus.indices": False,
+                "prometheus.cluster.settings": False,
+                "prometheus.nodes.filter": "_all",
```

Comment:
COS should collect metrics from each host individually, so only local metrics should be included.
```suggestion
                "prometheus.nodes.filter": "_local",
```

Reply:
Moved to snap: https://github.com/canonical/opensearch-snap/pull/47